api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -25,6 +25,11 @@ service RateLimitService {
   // Determine whether rate limiting should take place.
   rpc ShouldRateLimit(RateLimitRequest) returns (RateLimitResponse) {
   }
+
+  // Determine whether rate limiting should take place (Bidi-Stream RPC version).
```

Comment:
What are the semantics of this bidi stream?  For example, can multiple requests be in flight at the same time on the same stream?  If so, does the protocol provide enough information for the client to know which response is associated with which request?

Reply:
Good questions! I have some initial implementation.

> can multiple requests be in flight at the same time on the same stream? 

More specially, for sending message, either `GoogleAsyncStreamImpl::sendMessageRaw()` or `AsyncStreamImpl::sendMessage()` will be invoked based on use cases. The send is invoked [here](https://github.com/envoyproxy/envoy/blob/a5927dc3f59e79772a6007b3b5538e86b2a597b0/source/common/grpc/typed_async_client.cc#L15). 

AFAIU, only one request (one `envoy::service::ratelimit::v3::RateLimitRequest`, serialized message) will be sent at one time, but multiple requests (not at the same time) can be sent as long as stream is opened. 

Does this answer your questions? Also, I would like to hear your thoughts on this. Thanks!